clients/src/main/java/org/apache/kafka/common/serialization/Serialization.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common.serialization;
+
+/**
+ * The interface for wrapping a serializer and deserializer for the given data type.
+ *
+ * @param <T>
+ */
+public interface Serialization<T> {
+
+    Serializer<T> serializer();
```

Comment:
Question (@ijuma I'd be happy to hear your feedback, too):  Why do we keep having both `Serializer<T>` and `Deserializer<T>`, and then combine these to a `Serialization<T>`?  That's more boilerplate for developers that need to implement their own serdes.

Why not (and this is a question, I'm not implying a yes/no answer) something like this, which basically copies the current `Serializer<T>` interface but also includes a corresponding `deserialize` method:

```java
public interface Serde<T> extends Closeable {

    public void configure(Map<String, ?> configs, boolean isKey);
    
    public byte[] serialize(String topic, T data);

    public T deserialize(String topic, byte[] data);

    @Override
    public void close();
}
```

One difference of the above, "merged" approach would be that there's is just a single `configure` and `close` method for both the serialization and the deserialization functionality.